lib/kits/core-ui/paged-publication.js

```diff
@@ -0,0 +1,224 @@
+        ),
+        removedComponents: scriptEl.getAttribute(
+            'data-component-paged-publication-remove-components'
+        ),
+        theme: scriptEl.getAttribute('data-component-theme'),
+        offerClickBehavior: scriptEl.getAttribute(
+            'data-component-paged-publication-viewer-offer-click-behavior'
+        )
+    };
+
+    const customTemplates = {
+        mainContainer: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-template'
+        ),
+        offerList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-offer-list-template'
+        ),
+        shoppingList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-shopping-list-template'
+        ),
+        shoppingListCounter: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-shopping-list-counter-template'
+        ),
+        pageList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-page-list-template'
+        ),
+        menuContainer: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-menu-container-template'
+        )
+    };
+
+    const mainContainerEl = document.querySelector(scriptEls.mainContainer);
+
+    MainContainer({
+        template: customTemplates.mainContainer,
+        shoppingListCounterTemplate: customTemplates.shoppingListCounter,
+        el: mainContainerEl,
+        scriptEls
+    }).render();
+
+    const render = async () => {
+        if (Object.keys(options).length === 0) {
+            await setOptions();
+        }
+
+        start();
+
+        renderShoppingList();
+        renderMenuPopup();
+    };
+
+    const renderShoppingList = () => {
+        ShoppingList({
+            template: customTemplates.shoppingList
+        }).render();
+    };
+
+    const renderMenuPopup = () => {
+        const menuBtn = document.querySelector('.sgn__nav-menu-btn');
+        menuBtn.addEventListener('click', (e) => {
+            e.preventDefault();
+            MenuPopup({
+                configs: options,
+                sgnData,
+                sgnViewer,
+                templates: customTemplates
+            }).render();
+        });
+    };
+
+    const animateShoppingListCounter = () => {
+        const shoppingListCounter = document.querySelector(
+            '.sgn__offer-shopping-list-counter'
+        );
+
+        shoppingListCounter?.classList.remove('sgn-animate-bounce');
+        shoppingListCounter?.classList.add('sgn-animate-bounce');
+    };
+
+    const setOptions = async (opts = {}) => {
+        options = {
+            el: document.querySelector('.sgn__pp'),
+            eventTracker: SGN.config.get('eventTracker'),
+            pageId:
+                opts.pageId ||
+                `page${SGN.util.getQueryParam(scriptEls.pageIdParam) || 1}`,
+            id:
+                opts.id ||
+                SGN.util.getQueryParam(scriptEls.publicationIdParam) ||
```

Comment:
Please import these two SGN utilities directly from `../../util` instead, it is best we avoid using this singleton internally.